airflow/models/dag_version.py

```diff
@@ -136,7 +136,8 @@ def get_latest_version(cls, dag_id: str, *, session: Session = NEW_SESSION) -> D
     def get_version(
         cls,
         dag_id: str,
-        version_number: int = 1,
+        version_number: int | None = None,
+        version_name: str | None = None,
```

Comment:
@jedcunningham, I remember we removed the version_name, but this seems necessary when using the API & UI to look up the code version. Let me know what you think.

Reply:
I think it should either require it to be singular `get_version(version_number)`, or plural `get_versions(version_name)`. As proposed, getting the latest version for `get_version(version_name)` feels like a weird interface to me.

Reply:
From the UI perspective, how should users search for DAG version? Should it be dag_id and number only?

Reply:
What's the difference between the version number and name? Whatever is more human-readable is what the UI should use.

Reply:
The version number is incremented each time there's a structural change to the DAG, while the version name does not change. You can have all the DAGs with the same version name. I think for display, having `version_name-version_number` would be good. However, I'm wondering if we really need the version_name @jedcunningham.

Reply:
version_name is just a way for authors to add something that is meaningful for them. Strictly speaking we could do away with it completely.

Reply:
I have removed the version_name altogether.